Public/Src/Utilities/UnitTests/Storage/InputChangeListTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using BuildXL.Storage.ChangeTracking;
+using BuildXL.Storage.InputChange;
+using BuildXL.Utilities.Instrumentation.Common;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Test.BuildXL.Storage
+{
+    public sealed class InputChangeListTests : XunitBuildXLTest

```

Comment:
You need a few error case tests as well...

""
"|"
"relative"
"relative|BAD"

etc